### PR TITLE
Port handler-proxy (http-only)

### PR DESCRIPTION
**Pull request by maksimkim**
Original issue: https://github.com/netty/netty/issues/1133
Original PR: https://github.com/netty/netty/pull/2817

Motivation:
There is no support for client socket connections via a proxy server in SpanNetty.

Modifications: 
- Add a new module 'handler-proxy'
- Add ProxyHandler and its subclasses to support and HTTP(S) (no SOCKS by now) proxy connections
- Add a full parameterized test for most scenarios

Result:
A user can make an outgoing connection via proxy servers with only trivial effort.

### CODE REVIEW

File: test/DotNetty.Handlers.Proxy.Tests/ProxyHandlerTest.cs [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+        // look for "Seed used: *" debug message in test logs
+        private static readonly int ReproducibleSeed = 0;
+
+        public ProxyHandlerTest()
+        {
+            ClearServerExceptions();
+        }
+        
+        [Theory]
+        [MemberData(nameof(CreateTestItems))]
```
Comment (cuteant) [inline test/DotNetty.Handlers.Proxy.Tests/ProxyHandlerTest.cs:86]: https://dev.azure.com/SpanNetty/SpanNetty/_build/results?buildId=340&view=ms.vss-test-web.build-test-results-tab&runId=1576&resultId=107667&paneView=debug

and also test fail locally on Windows

Comment (maksimkim) [reply test/DotNetty.Handlers.Proxy.Tests/ProxyHandlerTest.cs:90]: fixed

**Comment by cuteant** [comment]
@maksimkim Also, some test cases(in 'libuv/handler' module) failed on azure devops, can be ignored for now(pass locally on Windows/Ubuntu)，we could fix them by #51

**Review by cuteant — APPROVED** [review]
LGTM 👍

**Pull request merged**